org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/ManifestMergerTest.java

```diff
@@ -16,20 +16,64 @@
 import java.util.LinkedHashSet;
 import java.util.jar.Attributes;
 import java.util.jar.Manifest;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.eclipse.xtext.util.MergeableManifest;
 import org.eclipse.xtext.util.StringInputStream;
 import org.eclipse.xtext.util.Strings;
 import org.eclipse.xtext.util.Wrapper;
 import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
 
 /**
  * @author Sven Efftinge - Initial contribution and API
  */
 @SuppressWarnings("deprecation")
 public class ManifestMergerTest extends Assert {
 	private static final String NL = Strings.newLine();
+	
+	private static final boolean isJava16OrLater = determineJava16OrLater();
+	
+	private static boolean determineJava16OrLater() {
+		String javaVersion = System.getProperty("java.version");
+		try {
+			Pattern p = Pattern.compile("(\\d+)(.)*");
+			Matcher matcher = p.matcher(javaVersion);
+			if (matcher.matches()) {
+				String first = matcher.group(1);
+				int version = Integer.parseInt(first);
+				return version >= 16;
+			}
+		} catch (NumberFormatException e) {
+			// ok
+		}
+		return false;
+	}
+	
+	@Rule
+	public MethodRule ignoreOnJava16OrLater = new MethodRule() {
```

Comment:
I like the idea a lot. JUnit5 has something like `@EnabledOnJre(..)` / `@DisabledOnJre` respectively. We could try to mimic that (if we need similar test-toggles in other classes, too).